src/core/server/http/http_server.test.ts

```diff
@@ -1409,13 +1410,93 @@ describe('setup contract', () => {
   });
 
   describe('#registerStaticDir', () => {
+    const assetFolder = join(__dirname, 'integration_tests', 'fixtures', 'static');
+
     test('does not throw if called after stop', async () => {
       const { registerStaticDir } = await server.setup(config);
       await server.stop();
       expect(() => {
         registerStaticDir('/path1/{path*}', '/path/to/resource');
       }).not.toThrow();
     });
+
+    test('returns correct headers for static assets', async () => {
```

Comment:
Should we collocate all the caching tests in a single place https://github.com/elastic/kibana/blob/master/test/server_integration/http/platform/cache.ts ?

Reply:
Would make sense. Would have to move the tests in the `plugin_functional` suite though, as we'll need some custom plugin to cover the test cases of this PR